tests/unit-tests/plugin-manager.php

```diff
@@ -0,0 +1,130 @@
+<?php
+/**
+ * Tests the plugin management functionality.
+ *
+ * @package Newspack\Tests
+ */
+
+use Newspack\Plugin_Manager;
+
+/**
+ * Test plugin management functionality.
+ */
+class Newspack_Test_Plugin_Manager extends WP_UnitTestCase {
+
+	/**
+	 * Plugin slug/folder.
+	 *
+	 * @var string
+	 */
+	protected $plugin_slug = 'hello-dolly';
+
+	/**
+	 * Plugin file path.
+	 *
+	 * @var string
+	 */
+	protected $plugin_file = 'hello-dolly/hello.php';
+
+	/**
+	 * URL to plugin download.
+	 *
+	 * @var string
+	 */
+	protected $plugin_url = 'https://downloads.wordpress.org/plugin/hello-dolly.1.6.zip';
+
+	/**
+	 * Compatibility checks and clean up.
+	 */
+	public function setUp() {
+		// These tests can't run on environments where we can't install plugins e.g. VIP Go.
+		if ( ! Plugin_Manager::can_install_plugins() ) {
+			$this->markTestSkipped( 'Plugin installation is not allowed in the environment' );
```

Comment:
Am I right in thinking this will skip _all_ tests if plugin (un-)install can't be performed? If so, we should probably make it conditional on the test, as not being able to (un-)install plugins doesn't preclude (de-)activation.

Reply:
It will skip all those tests if plugin installation is not allowed, even activation tests. The reason is because the activation tests can't run without first installing a plugin to test activation with. I'll investigate whether there's a good way of handling that situation.

Reply:
We can resolve the issue with our CI architecture, if e.g. Travis sets up a site with VIP Go configurations and a plugin preinstalled, the tests could be tweaked or we could have separate tests for that environment. I'll hold off on modifying this for now, though, since it wouldn't be super helpful without that environment.